dojo/tools/fortify/parser.py

```diff
@@ -114,6 +120,51 @@ def get_findings(self, filename, test):
                 dupes.add(title)
         return items
 
+    def parse_fpr(self, filename, test):
+        if str(filename.__class__) == "<class '_io.TextIOWrapper'>":
+            input_zip = zipfile.ZipFile(filename.name, 'r')
+        else:
+            input_zip = zipfile.ZipFile(filename, 'r')
+        zipdata = {name: input_zip.read(name) for name in input_zip.namelist()}
+        root = ElementTree.fromstring(zipdata["audit.fvdl"].decode('utf-8'))
+        regex = r"{.*}"
+        matches = re.match(regex, root.tag)
+        try:
+            namespace = matches.group(0)
+        except BaseException:
+            namespace = ""
+        items = list()
+        for child in root:
+            if "Vulnerabilities" in child.tag:
+                for vuln in child:
+                    ClassID = vuln.find(f"{namespace}ClassInfo").find(f"{namespace}ClassID").text
+                    Kingdom = vuln.find(f"{namespace}ClassInfo").find(f"{namespace}Kingdom").text
+                    Type = vuln.find(f"{namespace}ClassInfo").find(f"{namespace}Type").text
+                    AnalyzerName = vuln.find(f"{namespace}ClassInfo").find(f"{namespace}AnalyzerName").text
+                    DefaultSeverity = vuln.find(f"{namespace}ClassInfo").find(f"{namespace}DefaultSeverity").text
+                    InstanceID = vuln.find(f"{namespace}InstanceInfo").find(f"{namespace}InstanceID").text
+                    InstanceSeverity = vuln.find(f"{namespace}InstanceInfo").find(f"{namespace}InstanceSeverity").text
+                    Confidence = vuln.find(f"{namespace}InstanceInfo").find(f"{namespace}Confidence").text
+                    description = Type + "\n"
+                    description += "**ClassID:** " + ClassID + "\n"
+                    description += "**Kingdom:** " + Kingdom + "\n"
+                    description += "**AnalyzerName:** " + AnalyzerName + "\n"
+                    description += "**DefaultSeverity:** " + DefaultSeverity + "\n"
+                    description += "**InstanceID:** " + InstanceID + "\n"
+                    description += "**InstanceSeverity:** " + InstanceSeverity + "\n"
+                    description += "**Confidence:** " + Confidence + "\n"
+                    items.append(
+                        Finding(
+                            title=Type + " " + ClassID,
+                            severity="High",
+                            static_finding=True,
+                            test=test,
+                            description=description,
+                            unique_id_from_tool=ClassID,
+                        )
+                    )
+        return items
```

Comment:
can we also populate the test entry with data from the report like:
- Version>front_end_20240220_629_9_1361_6018341</Version>
- CreatedTS date="2024-02-20" time="05:34:11"/>
- end = created + ScanTime value="215"/>
- description based on:

```
  <NumberFiles>58</NumberFiles>
  <LOC type="Line Count">2942</LOC>
  <SourceBasePath>/var/lib/jenkins/workspace/SDM_SockShop_front-end_master/src</SourceBasePath>
  <SourceFiles> as list of files
```

Reply:
I don't know if we can populate this info to the test itself. Maybe I don't understand exactly what you are talking about @schdief .

Reply:
when you import a new scan result this is related to a test in defect dojo, currently you only write data for the findings

https://URL/finding/39

but it belongs to a 

https://URL/test/7

and this test has metadata itself

<img width="806" alt="image" src="https://github.com/DefectDojo/django-DefectDojo/assets/10979159/34a2fde7-6377-40e0-ab51-4d5ad001d2cc">

here we can add more data based on the fpr like version, start and end and within the description of the test also the number of files, loc and the list of sourcefiles from which sourcebaspath

Reply:
Yes, please look at the parser, I can't affect the test, only the findings. The test is affected by your API calls.

Reply:
I can see that the current implementation cannot update the test, but the GUI uses this form
https://URL/engagement/3/import_scan_results

to import scan results, e. g. of fortify, and at the same time creates a test container for it, maybe this should be another ticket/pr, but if the info doesn't get parsed, it won't be able to be stored later on